catalog/core/catalog-core-api-impl/src/test/groovy/ddf/catalog/util/impl/ResultIterableSpec.groovy

```diff
@@ -188,38 +192,105 @@ class QueryResultPaginatorSpec extends Specification {
         Query queryMock = createQueryMock(1, 0)
         QueryRequest queryRequestMock = createQueryRequestMock(queryMock)
 
-        def resultIterator = new ResultIterable(catalogFramework, queryRequestMock).iterator()
+        def resultIterator = resultIterable(catalogFramework, queryRequestMock).iterator()
         def results = []
 
         when:
         1.upto(resultSize + 1, { results << resultIterator.next() })
 
         then:
-        1.upto(resultSize + 1, { results.get(it - 1).distanceInMeters == (Double) it })
+        1.upto(resultSize + 1, { assert results.get(it - 1).distanceInMeters == (Double) it })
+    }
+
+    def "Properly pages when default page size is used with no maxResults"() {
+        setup:
+        def totalResults = 25
+        def resultSize = 20
+
+        3 * catalogFramework.query(_ as QueryRequest) >> {
+            QueryRequest queryRequest -> buildQueryResponse(queryRequest, resultSize, totalResults)
+        } >> {
+            QueryRequest queryRequest -> buildQueryResponse(queryRequest, 5, totalResults)
+        } >> {
+            QueryRequest queryRequest -> buildQueryResponse(queryRequest, 0, totalResults)
+        }
+
+        Query queryMock = createQueryMock(1, 0)
+        QueryRequest queryRequestMock = createQueryRequestMock(queryMock)
+
+        ResultIterable resultIterable = resultIterable(catalogFramework, queryRequestMock)
+
+        when:
+        def results = resultIterable.stream().collect(Collectors.toList())
+
+
+        then:
+        results.size() == 25
```

Comment:
❓ If we keep this test, would it be safer and more consistent to verify that all the `distanceInMeters` values are as expected?